canvas/raster.go

```diff
@@ -35,40 +39,48 @@ func NewRaster(generate func(w, h int) image.Image) *Raster {
 // and height parameters passed to pixelColor.
 func NewRasterWithPixels(pixelColor func(x, y, w, h int) color.Color) *Raster {
 	return &Raster{
-		Generator: func(w, h int) image.Image {
```

Comment:
As this is the crux of the change it’s really an internal re-use that is being exposed through the Generator function.

I think this could be solved by creating an internal type that handles the image re-use without needing to expose the api change.

Reply:
Haven't tried yet, but I get the feeling that instead of passing generator as a function ptr, it might be more flexible if you passed in an object that implemented a generator interface.

That would certainly make custom implementations a lot easier to handle. 

Will have to give it a play first. Thats still an API change, and could be a can of worms.  Hoping to get some playtime sooner rather than later.